Documentation/gettingstarted/encryption.rst

```diff
@@ -0,0 +1,130 @@
+keys may be manually distributed but that is not shown here.
+
+.. note::
+
+    This is a beta feature. Please provide feedback and file a GitHub issue
+    if you experience any problems. Currently only supported in tunnel mode. 
+
+First step is to download the Cilium Kubernetes descriptor:
+
+.. tabs::
+  .. group-tab:: K8s 1.13
+
+    .. parsed-literal::
+
+      curl -LO \ |SCM_WEB|\/examples/kubernetes/1.13/cilium-ds.yaml
+
+  .. group-tab:: K8s 1.12
+
+    .. parsed-literal::
+
+      curl -LO \ |SCM_WEB|\/examples/kubernetes/1.12/cilium-ds.yaml
+
+  .. group-tab:: K8s 1.11
+
+    .. parsed-literal::
+
+      curl -LO \ |SCM_WEB|\/examples/kubernetes/1.11/cilium-ds.yaml
+
+  .. group-tab:: K8s 1.10
+
+    .. parsed-literal::
+
+      curl -LO \ |SCM_WEB|\/examples/kubernetes/1.10/cilium-ds.yaml
+
+  .. group-tab:: K8s 1.9
+
+    .. parsed-literal::
+
+      curl -LO \ |SCM_WEB|\/examples/kubernetes/1.9/cilium-ds.yaml
+
+  .. group-tab:: K8s 1.8
+
+    .. parsed-literal::
+
+      curl -LO \ |SCM_WEB|\/examples/kubernetes/1.8/cilium-ds.yaml
+
+First create a yaml file for the IPSec keys to be stored as a etcd secret.
+The 'cilium-ipsec-keys.yaml' listed below gives an example.
+
+.. parsed-literal::
+  apiVersion: v1
+  kind: Secret
+  metadata:
+    name: cilium-ipsec-keys
+  type: Opaque
+  stringData:
+
+Next we will generate the necessary IPSec keys which will be distributed as an
+etcd secret using the 'cilium-ipsec-keys.yaml' file. In this example we use
+AES-CBC with HMAC-256 (hash based authentication code), but any of the supported
+Linux algorithms may be used. To generate use the following
+
+.. parsed-literal::
+  KEY1=0x`dd if=/dev/urandom count=32 bs=1 2> /dev/null| xxd -p -c 64`
+  KEY2=0x`dd if=/dev/urandom count=32 bs=1 2> /dev/null| xxd -p -c 64`
+  echo "keys: \\"hmac(sha256) $KEY1 cbc(aes) $KEY2\\"" >> cilium-ipsec-keys.yaml
+
+From the 'cilium-ipsec-keys.yaml' file generate an etcd secret.
+
+.. parsed-literal::
+  kubectl -n kube-system create -f cilium-ipsec-keys.yaml
+
+The secret can be displayed with 'kubectl -n kube-system get secret' and will be
+listed as 'cilium-ipsec-keys'.
+
+.. parsed-literal::
+ $ kubectl -n kube-system get secrets
+ NAME                                             TYPE                                  DATA   AGE
+ cilium-ipsec-keys                                Opaque                                1      105m
+
+After the secret is built we can enable 'cilium-agent'.
+To enable 'cilium-agent' we use a patch file to update the configuration with the required
+cilium-agent options and included IPSec keys.
+
+.. parsed-literal::
+  metadata:
+    namespace: kube-system
+  spec:
+    template:
+      spec:
+        containers:
+        - name: cilium-agent
+          args:
+          - "--debug=$(CILIUM_DEBUG)"
+          - "--kvstore=etcd"
+          - "--kvstore-opt=etcd.config=/var/lib/etcd-config/etcd.config"
+          - "--enable-ipsec"
+          - "--ipsec-key-file=/etc/ipsec/keys"
```

Comment:
This is OK for now but we should use an environment variable instead going forward

Reply:
OK will add it to the list.